### PR TITLE
Add Google Great Code Dataset

### PR DESCRIPTION
**Pull request by abhishekkrthakur**

### CODE REVIEW

File: src/datasets/utils/mock_download_manager.py [hunk 1]
```diff
@@ -169,6 +170,9 @@ def create_dummy_data_dict(self, path_to_dummy_data, data_url):
 
     def create_dummy_data_list(self, path_to_dummy_data, data_url):
         dummy_data_list = []
+        # trick: if there are many shards named like `data.txt-000001-of-00300`, only use the first one
+        if all(bool(re.findall("[0-9]{3,}-of-[0-9]{3,}", url)) for url in data_url):
+            data_url = [data_url[0]] * len(data_url)
```
Comment (lhoestq) [inline src/datasets/utils/mock_download_manager.py:175]: added this trick to be able to have only 1 file per split instead of 300 @abhishekkrthakur

Comment (abhishekkrthakur) [reply src/datasets/utils/mock_download_manager.py:175]: nice!

Comment (abhishekkrthakur) [reply src/datasets/utils/mock_download_manager.py:175]: file is much smaller now.

**Review by lhoestq — APPROVED** [review]
LGTM thanks :)

**Pull request merged**